### PR TITLE
Close #575.  Detect lock and fixed statements and switch sections wit…

### LINKED ISSUE #575: Make one Use Braces analyzer
We currently have three: ifs, loops, and usings.  We need to add locks to the mix as well.

### PR DESCRIPTION
**Pull request by Hosch250**
…hout braces.

### CODE REVIEW

File: VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/General/MissingBraces/MissingBracesAnalyzer.cs [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+        private static readonly string Category = VSDiagnosticsResources.GeneralCategory;
+        private static readonly string Message = VSDiagnosticsResources.MissingBracesAnalyzerMessage;
+        private static readonly string Title = VSDiagnosticsResources.MissingBracesAnalyzerTitle;
+
+        internal static DiagnosticDescriptor Rule
+            => new DiagnosticDescriptor(DiagnosticId.MissingBraces, Title, Message, Category, Severity, true);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context) => context.RegisterSyntaxNodeAction(AnalyzeSymbol,
```
Comment (Vannevelj) [inline VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/General/MissingBraces/MissingBracesAnalyzer.cs]: You could just register separate `SyntaxNodeActions` -- then you don't have to do the iskind + cast check anymore which would greatly improve performance I think

Comment (Hosch250) [inline VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/General/MissingBraces/MissingBracesAnalyzer.cs]: OK.

File: VSDiagnostics/VSDiagnostics/VSDiagnostics/VSDiagnosticsResources.resx [hunk 2]
```diff
@@ -654,4 +654,14 @@
   <data name="TestMethodWithoutTestAttributeAnalyzerTitle" xml:space="preserve">
     <value>A method might be missing a test attribute.</value>
   </data>
+  <data name="MissingBracesAnalyzerMessage" xml:space="preserve">
+    <value>A{0} {1} statement should be written with braces</value>
+    <comment>{0} - A/An</comment>
+  </data>
```
Comment (Hosch250) [inline VSDiagnostics/VSDiagnostics/VSDiagnostics/VSDiagnosticsResources.resx]: What do you think about this?  It doesn't seem particularly nice to me.

Comment (Vannevelj) [inline VSDiagnostics/VSDiagnostics/VSDiagnostics/VSDiagnosticsResources.resx]: Use the entire word in the argument, that makes it clearer from a caller perspective. so `a` or `an`

**Comment by Hosch250** [comment]
This is ready to merge.

**Pull request merged**